Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailHelpers.azsli

```diff
@@ -257,46 +257,42 @@ void BlendDetailMaterial(inout DetailSurface surface, uint material1, uint mater
 {
     DetailSurface tempSurface;
     GetDetailSurfaceForMaterial(tempSurface, material1, detailUv);
+    WeightDetailSurface(tempSurface, weight * (1.0 - blend));
+    AddDetailSurface(surface, tempSurface);
     if (material2 != 0xFF)
     {
-        WeightDetailSurface(tempSurface, weight * blend);
-        AddDetailSurface(surface, tempSurface);
-
         GetDetailSurfaceForMaterial(tempSurface, material2, detailUv);
-        WeightDetailSurface(tempSurface, weight * (1.0 - blend));
-        AddDetailSurface(surface, tempSurface);
-    }
-    else
-    {
-        WeightDetailSurface(tempSurface, weight);
+        WeightDetailSurface(tempSurface, weight * blend);
         AddDetailSurface(surface, tempSurface);
     }
 }
 
 /*
 Populates a DetailSurface with material data gathered form the 4 nearest samples to detailMaterialIdUv. The weight
 of each detail material's contribution is calculated based on the distance to the center point for that sample (for
-instance, if detailMaterialIdUv falls perfectly inbetween all 4 samples, then each sample will be weighed at 25%).
+instance, if detailMaterialIdUv falls perfectly in-between all 4 samples, then each sample will be weighed at 25%).
 Each sample can have two different detail materials defined with a blend value to determine their relative contribution.
-The detailUv is used for samping the textures of each detail material.
+The detailUv is used for sampling the textures of each detail material.
 */
 bool GetDetailSurface(inout DetailSurface surface, float2 detailMaterialIdUv, float2 detailUv)
 {
     float2 textureSize;
     TerrainSrg::m_detailMaterialIdImage.GetDimensions(textureSize.x, textureSize.y);
 
-    float2 detailMaterialIdCoord = detailMaterialIdUv * textureSize + 0.5;
-    float2 detailMaterailIdUv = detailMaterialIdCoord;
-    float2 gatherWeight = frac(detailMaterialIdCoord);
+    float2 detailMaterialIdCoord = detailMaterialIdUv * textureSize; // uv -> pixel coordinate
 
-    // Adjust the gather weight for better interpolation. This helps avoid diamond-shaped artifacts in binlinear filtering.
-    gatherWeight = 3.0 * gatherWeight * gatherWeight - 2.0 * gatherWeight * gatherWeight * gatherWeight;
+    // detailMaterialIdCoord could be negative, so add textureSize to ensure it is positive
+    detailMaterialIdCoord += textureSize; 
+
+    // The detail material id texture wraps since the "center" point can be anywhere in the texture, so mod by texturesize
+    int2 detailMaterailIdTopLeft = int2(detailMaterialIdCoord) % textureSize;
+    int2 detailMaterailIdBottomRight = (int2(detailMaterialIdCoord) + 1) % textureSize;
```

Comment:
Nit: "Materail" -> "Material"

Also, I don't have a better suggestion, but somehow I find "TopLeft" / "BottomRight" a bit less readable than before with the straight hard-coded 0 and 1 numbers, when reading lines like "TopLeft.x, BottomRight.y".  The names makes *sense*, I just really have to think about what I'm reading to parse that back into "Left, Bottom".  Just mentioning it in case it sparks any ideas.